atc/exec/check_step.go

```diff
@@ -82,6 +86,22 @@ func (step *CheckStep) Run(ctx context.Context, state RunState) error {
 		return fmt.Errorf("timeout parse: %w", err)
 	}
 
+	if step.policyChecker != nil {
```

Comment:
This makes sense. After read through the container launch process, I now choose to hook the check in function `FindOrCreateContainer` of `atc/worker/worker.go`. This way, there is no need to add the check to every individual step. Please see if you feel ok with that. (P.S. I haven't added tests for the change in worker.go yet, because I want to get your confirm with the direction first.)